drivers/dma/dma_intel_adsp_hda.c

```diff
@@ -454,15 +454,26 @@ void intel_adsp_hda_dma_isr(void)
 		cfg = host_dev[i]->config;
 
 		for (j = 0; j < dma_ctx->dma_channels; j++) {
-			if (atomic_test_bit(dma_ctx->atomic, j)) {
-				clear_l1_exit |=
-					intel_adsp_hda_check_buffer_interrupt(cfg->base,
-									      cfg->regblock_size,
-									      j);
+			if (!atomic_test_bit(dma_ctx->atomic, j))
+				continue;
+
+			if (!intel_adsp_hda_is_buffer_interrupt_enabled(cfg->base,
+									cfg->regblock_size, j))
+				continue;
+
+			if (intel_adsp_hda_check_buffer_interrupt(cfg->base,
+								  cfg->regblock_size, j)) {
+				clear_l1_exit = true;
 				intel_adsp_hda_disable_buffer_interrupt(cfg->base,
 									cfg->regblock_size, j);
 				intel_adsp_hda_clear_buffer_interrupt(cfg->base,
 								      cfg->regblock_size, j);
+			} else {
+				/*
+				 * Postpone entering L1 state until all enabled interrupts
+				 * arrived, i.e. transfer started on all channels.
```

Comment:
did you mean 'transfer completed on all channels'?

Not sure I get the point of entering L1 as soon as transfers start.

Reply:
To initiate transfer, DSP must be in L0 state. Once the transfer is started, DSP can go to the low power L1 state, and the transfer will be able to continue and finish in L1 state. Interrupts are configured to trigger after the first 32 bytes of data arrive. Once such an interrupt arrives, the transfer has already started. If all expected transfers have started, it is safe to allow the low power L1 state.

Reply:
Added comment with the explanation.